admin/settings/class-abstract-settings.php

```diff
@@ -0,0 +1,443 @@
+<?php
+
+// If this file is called directly, abort.
+defined('WPINC') || die;
+
+abstract class LNP_SettingsPage
+{
```

Comment:
we could name it something like `BaseSettings` or something like that. I would suggest the class name be similar to its filename